tests/psbt.rs

```diff
@@ -0,0 +1,702 @@
+fn signer_two_sign(psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let expected_psbt_hex = include_str!("data/sign_2_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    let psbt = sign(psbt, keys);
+
+    assert_eq!(psbt, expected_psbt);
+    psbt
+}
+
+/// Does the combine according to the BIP, returns the combined PSBT. Verifies against BIP 174 test vector.
+fn combine(mut this: Psbt, that: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/combine_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    this.combine(that).expect("failed to combine PSBTs");
+
+    assert_eq!(this, expected_psbt);
+    this
+}
+
+/// Does the finalize step according to the BIP, returns the combined PSBT. Verifies against BIP 174
+/// test vector.
+fn finalize(psbt: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/finalize_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    let psbt = finalize_psbt(psbt);
+
+    assert_eq!(psbt, expected_psbt);
+    psbt
+}
+
+/// Does the transaction extractor step according to the BIP, returns the combined PSBT. Verifies
+/// against BIP 174 test vector.
+fn extract_transaction(psbt: Psbt) -> Transaction {
+    let expected_tx_hex = include_str!("data/extract_tx_hex");
+
+    let tx = psbt.extract_tx();
+
+    let got = serialize_hex(&tx);
+    assert_eq!(got, expected_tx_hex);
+
+    tx
+}
+
+/// Combines two PSBTs lexicographically according to the BIP. Verifies against BIP 174 test vector.
+fn combine_lexicographically() {
+    let psbt_1_hex = include_str!("data/lex_psbt_1_hex");
+    let psbt_2_hex = include_str!("data/lex_psbt_2_hex");
+
+    let expected_psbt_hex = include_str!("data/lex_combine_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+    
+    let v = Vec::from_hex(psbt_1_hex).unwrap();
+    let mut psbt_1: Psbt = deserialize(&v).expect("failed to deserialize psbt 1");
+
+    let v = Vec::from_hex(psbt_2_hex).unwrap();
+    let psbt_2: Psbt = deserialize(&v).expect("failed to deserialize psbt 2");
+
+    psbt_1.combine(psbt_2).expect("failed to combine PSBTs");
+    
+    assert_eq!(psbt_1, expected_psbt);
+}
+
+/// Signs `psbt` with `keys` if required.
+fn sign(mut psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let secp = SECP256K1;
+
+    let mut inputs = psbt.inputs.clone();
+
+    for (i, input) in inputs.iter_mut().enumerate() {
+        for sk in keys {
+            let mut required_pks = input.bip32_derivation.keys();
+            let pk = sk.public_key(secp);
+            if required_pks.any(|required| (*required) == pk.inner) {
+                psbt_sign::sign(&mut psbt, sk, i, secp).expect("failed to sign input with key");
+            }
+        }
+    }
+
+    psbt
+}
+
+/// Finalizes a PSBT accord to the Input Finalizer role described in BIP 174.
```

Comment:
Perhaps we should comment here that finalizing can better be done with `miniscript`?

Reply:
If we do we should add the same comment in `examples/ecdsa-psbt.rs`.